apm-agent-core/src/test/java/co/elastic/apm/agent/impl/ElasticApmTracerTest.java

```diff
@@ -56,17 +62,24 @@
     private MockReporter reporter;
     private ConfigurationRegistry config;
 
+    private ApmServerClient apmServerClient;
+
     private TestObjectPoolFactory objectPoolFactory;
 
     @BeforeEach
     void setUp() {
         objectPoolFactory = new TestObjectPoolFactory();
         reporter = new MockReporter();
         config = SpyConfiguration.createSpyConfig();
+
+        apmServerClient = new ApmServerClient(config.getConfig(ReporterConfiguration.class), config.getConfig(CoreConfiguration.class));
+        apmServerClient = mock(ApmServerClient.class, delegatesTo(apmServerClient));
```

Comment:
[for reviewer] this allows to delegate on default implementation and also enables us to mock some methods for easier testing.